ruby assignments program

RakeshH-25 opened the pull request:
sir i have pushed ruby files please have a look at it and in triangle programs i could not align the triangle perfectly so please check once and guide me if there needs some improvement in code

RakeshH-25 committed:
ruby assignments program

RakeshH-25 committed:
modified to get result as array

manoj-v2 commented:
I haven't tried running all the code.   All I can say is that.. you should start thinking in Ruby and not in Java.   Your program syntax is Ruby.. but  algorithm etc is still in Java.

RakeshH-25 committed:
changed code( program takes cmd line args) and added testing code

RakeshH-25 commented:
sir
i have modified 2 triangle programs which takes no. of lines as command line arguments for triangle and also added testing code and the 2 parameters for method are line-no and total no of lines in the triangle

RakeshH-25 committed:
Modified and new files

RakeshH-25 commented:
hello sir,
i have tried the first file assignment with the name FileExample.
the problem is, the program produces the required o/p but its failing in testing, can you please have a look on it and tell how can i overcome it or how to better it.

RakeshH-25 committed:
rectified the error

RakeshH-25 commented:
i removed the puts in method and its working fine now (testing)

manoj-v2 commented on FileExample.rb:
```diff
@@ -0,0 +1,31 @@
+	
+	def get_day(path)
+		@col3 = []
+		@col1 = []
```
These 2 variables need not be instance variables.  They are only used in this method. So you can make it local variable.

manoj-v2 commented on FileExample.rb:
```diff
@@ -0,0 +1,31 @@
+		File.open(path, "r").each_line do |line|
+		  	#puts line
+		  	data = line.split(/\s+/)
+		  	 @col3 << data[3] if (data[3] != nil && data[3].to_i > 0)
```
data[3] represent 4th column ?  or 3rd column

RakeshH-25 commented:
its column header is MnT

manoj-v2 commented on FileExample.rb:
```diff
@@ -0,0 +1,31 @@
+		  	#puts line
+		  	data = line.split(/\s+/)
+		  	 @col3 << data[3] if (data[3] != nil && data[3].to_i > 0)
+		  	 @col1 << data[1] if (data[3] != nil && data[3].to_i > 0)
```
why is data[3] compared for nil here ?  should it be data[1] ?

RakeshH-25 commented:
now i have one more doubt sir the requirement is finding min of Mnt or min of Mnt-Mxt? my solution is for min of Mnt

RakeshH-25 commented:
sir, because there is an empty line between header and data line, i was encountering nil error

RakeshH-25 committed:
Modified as per comments

RakeshH-25 committed:
for MxT-MnT

RakeshH-25 commented:
sir now the program runs to find min(MxT-MnT)

manoj-v2 commented on FileExample.rb:
```diff
@@ -0,0 +1,29 @@
+class TestNumbers < Test::Unit::TestCase
+    def test_line
+    	file="/home/rakesh/Downloads/weather.txt"
+        assert_equal(22, Whether.new.get_day(file))
```
Can you please explain why it should be 22 ?

RakeshH-25 committed:
modified file

RakeshH-25 commented:
gm sir,
After cross-verifying i have found that my logic should return 14 instead of 22, i am sorry for pushing wrong one, from next time i'll cross-verify before pushing it.

RakeshH-25 committed:
Ruby Program for Gradle file

RakeshH-25 committed:
Ruby Program for Gradle file

RakeshH-25 commented:
hello sir,
 i have pushed 2 files for replacing values (in fact it replaces the complete line) in gradle files as per the mail details,please check and tell me is this the correct way to do that and how can i improve my code. and can you please guide me how can i write test for the program

manoj-v2 commented on gradlefile1.rb:
```diff
@@ -0,0 +1,48 @@
+		  		if line[/versionName/] and name_sub == false
+		  			p line
+		  			name_sub_line = line
+		  			substitute(path,name_sub_line,"versionName = \"#{version_name}\"\n")
```
In one of the gradle file,  there was no equal to  after versionName or versionCode.  In this case you are force fully adding the '='.  If the '=' is not there, then you should not add it.

RakeshH-25 replied:
that means should i write only one program which can take both files and does as you commented?

RakeshH-25 committed:
Silgle file to handle both gradle file

RakeshH-25 commented:
sir,
Now gradlefile1.rb is updated and please giude me if my code need improvement

RakeshH-25 committed:
New solution for gradle files

RakeshH-25 commented:
sir,
i got a new solution for gradle files and the file name is newgradlefile.rb
